### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by Bombaleo**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # rep-for-links
 Repository for links
+Link on heroku: https://rep-for-urls.herokuapp.com/
```
Comment (anatoliliotych) [inline README.md]: link to

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+SIGN_IN_ERROR = 'This user does not exist, please authorizate.'.freeze
+
+use Rack::Session::Pool, cookie_only: false
+
+def authenticated?
+  session[:username]
+end
+
+get '/' do
+  if authenticated?
```
Comment (anatoliliotych) [inline app.rb]: make this method smaller by if in one line

File: app.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    redirect '/submit'
+  else
+    erb :auth, locals: { client_id: CLIENT_ID }
+  end
+end
+
+get '/callback' do
+  session_code = request.env['rack.request.query_hash']['code']
+  result = RestClient.post('https://github.com/login/oauth/access_token',
```
Comment (anatoliliotych) [inline app.rb]: move url to constant

File: app.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                        { client_id: CLIENT_ID,
+                         client_secret: CLIENT_SECRET,
+                         code: session_code },
+                         accept: :json)
+  session[:access_token] = JSON.parse(result)['access_token']
+  access_token = session[:access_token]
+  begin
+    auth_result = RestClient.get('https://api.github.com/user',
```
Comment (anatoliliotych) [inline app.rb]: url to constant

File: app.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                                 { params: { access_token: access_token },
+                                  accept: :json })
+  rescue => e
+    puts e
```
Comment (anatoliliotych) [inline app.rb]: use logger

File: app.rb [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+  return (erb :error, locals: { message: SIGN_IN_ERROR }) unless @user
+  if @user.save
+    session[:username] = @user.name
+    erb :index
+  else
+    'Sorry, there was an error!'
+  end
+end
+
+post '/delurl' do
```
Comment (anatoliliotych) [inline app.rb]: use proper naming

File: app.rb [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+  @user = User.find_by(name: params[:name])
+  @user.urls.find_by(link: params[:link]).destroy
+  erb :index
+end
+
+post '/addurl' do
```
Comment (anatoliliotych) [inline app.rb]: use proper naming

File: config/database.yml [hunk 8]
```diff
@@ -0,0 +1,6 @@
+development:
+  adapter: postgresql
+  database: develop
+  username: <%= ENV['PG_USER'] %>
+  password: <%= ENV['PG_PASS'] %>
+host: localhost
```
Comment (anatoliliotych) [inline config/database.yml]: check formatting

File: config/environments.rb [hunk 9]
```diff
@@ -0,0 +1,15 @@
+
+#The environment variable DATABASE_URL should be in the following format:
+# => postgres://{user}:{password}@{host}:{port}/path
+configure :production, :develop do
+	db = URI.parse(ENV['DATABASE_URL'] || 'postgres://localhost/fsp_db')
+
+	ActiveRecord::Base.establish_connection(
+			:adapter => db.scheme == 'postgres' ? 'postgresql' : db.scheme,
```
Comment (anatoliliotych) [inline config/environments.rb]: check style

File: db/migrate/20170717164327_create_user.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
+
+class CreateUser < ActiveRecord::Migration[5.1]
+  def change
+  	create_table :users do |t|
+  		t.string :name
+      t.string :password
```
Comment (anatoliliotych) [inline db/migrate/20170717164327_create_user.rb]: check style

File: views/auth.erb [hunk 11]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+	<head>
```
Comment (anatoliliotych) [inline views/auth.erb]: check style

File: views/error.erb [hunk 12]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html>
+	<head>
+		 <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+	</head>
+	<body>
+		<div>
+	    <div  align="center">
+				<h3>
```
Comment (anatoliliotych) [inline views/error.erb]: check style

File: views/index.erb [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+		    background: #20B2AA	;
+	    	padding: 10px;
+		    border-radius: 5px;
+				white-space: normal;
+		   }
+			 .div1 {
+				 	display:inline-block;
+					overflow-x:hidden;
+				 	padding: 10px;
+	 	    	border-radius: 5px;
```
Comment (anatoliliotych) [inline views/index.erb]: check style

File: views/signup.erb [hunk 14]
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<html>
+	<head>
+		 <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+		 <script>
+			 function check_pass() {
+				 if (document.getElementById('password').value ==
+							document.getElementById('confirm_password').value) {
```
Comment (anatoliliotych) [inline views/signup.erb]: check style

File: views/signup.erb [hunk 15]
```diff
@@ -0,0 +1,57 @@
+<!DOCTYPE html>
+<html>
+	<head>
+		 <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+		 <script>
+			 function check_pass() {
```
Comment (anatoliliotych) [inline views/signup.erb]: move js out of the erb

**Comment by Bombaleo** [comment]
All done. Please review again.

File: views/auth.erb [hunk 16]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
+</head>
+<body>
+	<div align="center">
+		<div>
+			<h3>
+				Hey. Here you can leave your links to various sites.
```
Comment (anatoliliotych) [inline views/auth.erb]: check style!

File: views/error.erb [hunk 17]
```diff
@@ -0,0 +1,27 @@
...
+			</h3>
+			<div>
+				<h4> Go to registration page</h4>
+				<form method="GET" action="/signup">
+					<input type="submit" value="Signup">
+				</form>
+			</div>
+			<div>
+				<h4> Try to log in</h4>
+				<form method="GET" action="/">
```
Comment (anatoliliotych) [inline views/error.erb]: check style!

File: views/index.erb [hunk 18]
```diff
@@ -0,0 +1,49 @@
...
+							<p>
+								<%= elem.description %>
+							</p>
+						</div>
+					</td>
+					<td>
+						<div  id="right" class="div2">
+							<form  method="POST" action="/deleteurl">
+								<input type="hidden" name="link" value=<%= elem.link %> >
+								<input type="hidden" name="name" value=<%= @user.name %> >
```
Comment (anatoliliotych) [inline views/index.erb]: check style

File: views/signup.erb [hunk 19]
```diff
@@ -0,0 +1,44 @@
...
+		</h3>
+	</div>
+	<form action="/signup" method="POST">
+		<table>
+			<tr>
+				<td>
+					<label>name:</label>
+				</td>
+				<td>
+					<input type="text" name="user[name]" required value="" placeholder="Enter name" />
```
Comment (anatoliliotych) [inline views/signup.erb]: check style

File: config/environments.rb [hunk 20]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (anatoliliotych) [inline config/environments.rb]: remove

File: db/migrate/20170717164327_create_user.rb [hunk 21]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (anatoliliotych) [inline db/migrate/20170717164327_create_user.rb]: remove

File: db/migrate/20170717232205_add_reference_url.rb [hunk 22]
```diff
@@ -0,0 +1,6 @@
+class AddReferenceUrl < ActiveRecord::Migration[5.1]
+  def change
+    add_reference :urls, :user, foreign_key: 'name'
+  end
+end
+
```
Comment (anatoliliotych) [inline db/migrate/20170717232205_add_reference_url.rb]: remove

**Pull request merged**